projects/core/src/main/resources/data/computercraft/lua/rom/programs/shell.lua

```diff
@@ -54,6 +54,68 @@ else
     bgColour = colours.black
 end
 
+local function tokenise(...)
+    local sLine = table.concat({ ... }, " ")
+    local tWords = {}
+    local bQuoted = false
+    for match in string.gmatch(sLine .. "\"", "(.-)\"") do
+        if bQuoted then
+            table.insert(tWords, match)
+        else
+            for m in string.gmatch(match, "[^ \t]+") do
+                table.insert(tWords, m)
+            end
+        end
+        bQuoted = not bQuoted
+    end
+    return tWords
+end
+
+-- Execute a program using os.run, unless a shebang is present.
+-- In that case, execute the program using the interpreter specified in the shebang.
+-- This may occur recursively, up to the maximum number of times specified by nRemainingRecursion
+-- Returns the same type as os.run, which is a boolean indicating whether the program exited successfully.
+local function executeProgram(nRemainingRecursion, env, sPath, ...)
+    local file = fs.open(sPath, "r")
+    if not file then
+        printError("File not found: " .. sPath)
+        return false
+    end
+
+    -- First check if the file begins with a #!
+    local sContents = file.readLine()
```

Comment:
I cannot wait until Lua 5.4 when we have `<close>`.

```suggestion
    local sContents = file.readLine()
    file.close()
```

Also, would you be able to drop the Hungrarian notation in this function? Slowly trying to transition the codebase to not use it, it's just a slow process :).

Reply:
Oops, thanks. And sure, I wasn't aware we were trying to phase it out :P